### PR TITLE
Pdf2Text

### LINKED ISSUE #608: PDF2text
# Aim

- What is the objective of the Script
To convert PDF to Text
# Details

- What the features will your script have
My script converts PDF to Text

### PR DESCRIPTION
**Pull request by Amit366**
# Description

Converts a pdf file to text file

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by Kushal997-das — CHANGES_REQUESTED** [review]
@Amit366 if i put my path like the below it will give me the error . It occurs because i used This a normal string as a path. To overcome this just put `r` before your normal string it converts normal string to raw string. So change this so that everyone can use your project . 

gssoc1 (2)

Moreover , there could be more error i think  . Please @kaustubhgupta check this carefully thank you :)

**Comment by Amit366** [comment]
@Kushal997-das on  running the script it asks the user to enter the url. But as I can see you have changed input part.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Why there are flappy bird game files in this PR?

**Comment by Amit366** [comment]
@kaustubhgupta I am really sorry

**Comment by Amit366** [comment]
I will remove that

**Comment by kaustubhgupta** [comment]
> I will remove that

Don't remove them one by one. Just delete this branch, and create new PR from new branch

**Comment by Amit366** [comment]
@kaustubhgupta I will deleted and also commited it

**Comment by Kushal997-das** [comment]
> @Kushal997-das on running the script it asks the user to enter the url . But as I can see you have changed input part.

It doesn't matter if you give input runtime or not it gives you same error as same input .

**Comment by Amit366** [comment]
@Kushal997-das in run time it doesn't give the error

**Comment by Kushal997-das** [comment]
> > @Kushal997-das on running the script it asks the user to enter the url . But as I can see you have changed input part.
> 
> It doesn't matter if you give input runtime or not it gives you same error as same input .

> > @Kushal997-das in run time it doesn't give the error
>

> Could You please make a demo video of it ? And moreover in worst case scenario your program should run in both cases .

**Comment by Amit366** [comment]
@Kushal997-das I made the changes

**Comment by Kushal997-das** [comment]
> @Kushal997-das I made the changes

@Amit366 Yeah i saw . @kaustubhgupta Could you please run this code in your local because in my system this program is not working properly .

**Comment by Amit366** [comment]
@Kushal997-das is it showing any error??

File: PDF2Text/Readme.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+### Step 3:
+
+    Run the command: python script.py/python3 script.py 🧐
+
+### Step 4:
+
+    Sit back and Relax. Let the Script do the Job. ☕
+
+### Requirements
```
Comment (kaustubhgupta) [inline PDF2Text/Readme.md:27]:
```suggestion
## Requirements
```

File: PDF2Text/script.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import PyPDF2
+
+pdf = input(r"Enter the path of PDF file: ")
+n = int(input("Enter number of pages: "))
+st=""
+page = PyPDF2.PdfFileReader(pdf) 
+for i in range(n):
+    
+    st += page.getPage(i).extractText()
```
Comment (kaustubhgupta) [inline PDF2Text/script.py:9]: You need to store each page individually. That means, if I have a pdf of 100 pages, then 100 text files must be created, one file for one page of PDF.

File: PDF2Text/script.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+with open('./PDF2Text/pages/text.txt','w') as f:
```
Comment (kaustubhgupta) [inline PDF2Text/script.py:11]: This will not work as there is no folder named "pages".

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: PDF2Text/script.py (hunk 2 above)
Comment (Amit366) [reply PDF2Text/script.py:9]: @kaustubhgupta  can you suggest me the code for separately storing the pdf

Comment (kaustubhgupta) [reply PDF2Text/script.py:9]: as you are getting the page numbers from the user, you can write a for a loop until that number, then in that for loop, use the with-open pair of file handling to store that text as txt files and name them as 1,2,3 and so on. Naming will be easier as you can use current iteration number as file name (using f-strings)

**Comment by Amit366** [comment]
@kaustubhgupta made the changes

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
No text gets extracted from PDFs. Only blank test files were generated. Test the script locally and fix the bugs.

**Comment by Kushal997-das** [comment]
> No text gets extracted from PDFs. Only blank test files were generated. Test the script locally and fix the bugs.

Same problem i also faced @kaustubhgupta

**Comment by Amit366** [comment]
@kaustubhgupta it's working fine for me

**Comment by Amit366** [comment]
@kaustubhgupta do I thing. Place the PDF file into the script folder. Then try once.

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta do I thing. Place the PDF file into the script folder. Then try once.

It's not about the path, it' about: no text was present in the files. I tested this for my resume, the document with text only

**Comment by Amit366** [comment]
@kaustubhgupta then how is it working for me

**Comment by Amit366** [comment]
I will send the pics.

**Comment by kaustubhgupta** [comment]
> > No text gets extracted from PDFs. Only blank test files were generated. Test the script locally and fix the bugs.
> 
> Same problem i also faced @kaustubhgupta

I am not the only one

**Comment by Kushal997-das** [comment]
> I will send the pics.

Send one gif of this even in my case i faced same problem .

**Comment by Amit366** [comment]
Screenshot (127)

Screenshot (128)


Screenshot (129)

**Comment by Amit366** [comment]
Screenshot (131)

Screenshot (132)

**Comment by kaustubhgupta** [comment]
@Amit366 Okay got it. Include these Screenshots in the readme file too. Name that section: "Expected Output"

**Comment by Amit366** [comment]
@kaustubhgupta I have placed the pics into the readme

**Comment by Amit366** [comment]
@kaustubhgupta please review it.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Images not rendered: 

image

**Comment by Amit366** [comment]
@kaustubhgupta please review it

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by Kushal997-das — APPROVED** [review]
LGTM

**Comment by Amit366** [comment]
@kaustubhgupta it was a level 2 issue but it is given level 1. Please check.

**Comment by Amit366** [comment]
@kaustubhgupta can u please check the level

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta can u please check the level

Yes, the level was initially 2 but as you submitted the code, 3 days ago, we changed it to level 1 as the code doesn't work for all types of files plus the Code Quality was not up to level2

**Comment by Amit366** [comment]
@kaustubhgupta if the script is run in Vs code it will run fine .Once try it out

**Pull request merged**